discovery_engine/lib/src/domain/search_manager.dart

```diff
@@ -191,6 +194,40 @@ class SearchManager {
     return EngineEvent.activeSearchTermRequestSucceeded(search.searchTerm);
   }
 
+  /// Obtains the deep search documents related to `term` and `market`.
+  ///
+  /// These documents aren't persisted to repositories.
+  Future<EngineEvent> deepSearchRequested(
+    String term,
+    FeedMarket market,
+  ) async {
+    final List<DocumentWithActiveData> docs;
+    try {
+      docs = await _engine.deepSearch(term, market);
+    } catch (e) {
+      const fewWords =
+          'The sequence must contain at least `KEY_PHRASE_SIZE` valid words';
+      const notFound = 'HTTP status client error (404 Not Found) for url';
+      final message = e.toString();
+      if (message.contains(fewWords) || message.contains(notFound)) {
```

Comment:
❓ just wondering about "robustness" of checking for specific exception text... but I know we are probably limited in what we can do here...

Reply:
as long as we don't change the error messages on the rust side we should be fine, but i agree that this is not very robust. unfortunately all rust error types are just stringified on the dart side and we currently have no better way to handle this. it's probably also no worth to improve on the ffi error types now, since we move to protobufs.